### ISSUE TITLE
getConsignmentNote()->getResponseXml() is null

### ISSUE THREAD

**Issue by imjonos** [200MPH/tnt#17]
Hi 
I have this issue when i try do this
 $response->getActivity()->getConsignmentNote()->getResponseXml()

getErrors() 
return Response XML document is not valid or empty

![image](https://user-images.githubusercontent.com/5132976/77865333-6cf4bb80-7236-11ea-972b-2304bd19e207.png)

```
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<ESHIPPER>
 <LOGIN>
  <COMPANY></COMPANY>
  <PASSWORD></PASSWORD>
  <APPID>IN</APPID>
  <APPVERSION>3.0</APPVERSION>
 </LOGIN>
 <CONSIGNMENTBATCH>
  <SENDER><COMPANYNAME><![CDATA[11111111]]></COMPANYNAME>
<STREETADDRESS1><![CDATA[Meshherskiy pr-kt d 1/15 ]]></STREETADDRESS1>
<CITY><![CDATA[Moscow]]></CITY>
<POSTCODE><![CDATA[119618]]></POSTCODE>
<COUNTRY><![CDATA[RU]]></COUNTRY>
<ACCOUNT><![CDATA[000148713]]></ACCOUNT>
<CONTACTNAME><![CDATA[vapvpv vpvpav]]></CONTACTNAME>
<CONTACTDIALCODE><![CDATA[ ]]></CONTACTDIALCODE>
<CONTACTTELEPHONE><![CDATA[+7 999 999-99-99]]></CONTACTTELEPHONE>
<CONTACTEMAIL><![CDATA[moguchuhdima@gmail.com]]></CONTACTEMAIL>
   <COLLECTION><COLLECTIONADDRESS><COMPANYNAME><![CDATA[11111111]]></COMPANYNAME>
<STREETADDRESS1><![CDATA[Meshherskiy pr-kt d 1/15 ]]></STREETADDRESS1>
<CITY><![CDATA[Moscow]]></CITY>
<POSTCODE><![CDATA[119618]]></POSTCODE>
<COUNTRY><![CDATA[RU]]></COUNTRY>

<CONTACTNAME><![CDATA[vapvpv vpvpav]]></CONTACTNAME>
<CONTACTDIALCODE><![CDATA[ ]]></CONTACTDIALCODE>
<CONTACTTELEPHONE><![CDATA[+7 999 999-99-99]]></CONTACTTELEPHONE>
<CONTACTEMAIL><![CDATA[moguchuhdima@gmail.com]]></CONTACTEMAIL>
</COLLECTIONADDRESS>
<SHIPDATE><![CDATA[31/03/2020]]></SHIPDATE>
<PREFCOLLECTTIME>
 <FROM><![CDATA[09:03]]></FROM>
 <TO><![CDATA[18:03]]></TO>
</PREFCOLLECTTIME>
<COLLINSTRUCTIONS><![CDATA[]]></COLLINSTRUCTIONS>
</COLLECTION>
  </SENDER>
  <CONSIGNMENT><CONREF>ref01_008</CONREF>
<DETAILS>
 <RECEIVER><COMPANYNAME><![CDATA[Conan Doyle Ltd]]></COMPANYNAME>
<STREETADDRESS1><![CDATA[Baker Street 221b ]]></STREETADDRESS1>
<CITY><![CDATA[London]]></CITY>
<POSTCODE><![CDATA[NW1 6XE]]></POSTCODE>
<COUNTRY><![CDATA[GB]]></COUNTRY>
<CONTACTNAME><![CDATA[Sherlock Holmes]]></CONTACTNAME>
<CONTACTDIALCODE><![CDATA[ ]]></CONTACTDIALCODE>
<CONTACTTELEPHONE><![CDATA[+44 1234 567890]]></CONTACTTELEPHONE>
<CONTACTEMAIL><![CDATA[moguchuhdima@gmail.com]]></CONTACTEMAIL>
</RECEIVER>
 <DELIVERY><COMPANYNAME><![CDATA[Conan Doyle Ltd]]></COMPANYNAME>
<STREETADDRESS1><![CDATA[Baker Street 221b ]]></STREETADDRESS1>
<CITY><![CDATA[London]]></CITY>
<POSTCODE><![CDATA[NW1 6XE]]></POSTCODE>
<COUNTRY><![CDATA[GB]]></COUNTRY>
<CONTACTNAME><![CDATA[Sherlock Holmes]]></CONTACTNAME>
<CONTACTDIALCODE><![CDATA[ ]]></CONTACTDIALCODE>
<CONTACTTELEPHONE><![CDATA[+44 1234 567890]]></CONTACTTELEPHONE>
<CONTACTEMAIL><![CDATA[moguchuhdima@gmail.com]]></CONTACTEMAIL>
</DELIVERY>
<CUSTOMERREF><![CDATA[Weborder_123456]]></CUSTOMERREF>
<CONTYPE><![CDATA[D]]></CONTYPE>
<PAYMENTIND><![CDATA[S]]></PAYMENTIND>
<ITEMS><![CDATA[1]]></ITEMS>
<TOTALWEIGHT><![CDATA[0.5]]></TOTALWEIGHT>
<TOTALVOLUME><![CDATA[0.000918]]></TOTALVOLUME>
<SERVICE><![CDATA[09D]]></SERVICE>
<DESCRIPTION><![CDATA[ 
]]></DESCRIPTION>
<DELIVERYINST><![CDATA[Consulting Detective]]></DELIVERYINST>
 <PACKAGE><ITEMS><![CDATA[1]]></ITEMS>
<DESCRIPTION><![CDATA[Description]]></DESCRIPTION>
<LENGTH><![CDATA[0.01]]></LENGTH>
<HEIGHT><![CDATA[0.27]]></HEIGHT>
<WIDTH><![CDATA[0.34]]></WIDTH>
<WEIGHT><![CDATA[0.5]]></WEIGHT>
</PACKAGE>
</DETAILS>
</CONSIGNMENT>
 </CONSIGNMENTBATCH>
<ACTIVITY><CREATE>
 <CONREF><![CDATA[ref01_008]]></CONREF>
</CREATE>
<BOOK ShowBookingRef="Y">
 <CONREF><![CDATA[ref01_008]]></CONREF>
</BOOK>
<SHIP>
 <CONREF><![CDATA[ref01_008]]></CONREF>
</SHIP>
 <PRINT><REQUIRED>
 <CONREF><![CDATA[ref01_008]]></CONREF>
</REQUIRED>
</PRINT>
 <SHOW_GROUPCODE/>
</ACTIVITY>
</ESHIPPER>

```

If I send this
```
<PRINT>
         <CONNOTE>
            <CONREF><![CDATA[ref01_008]]></CONREF>
         </CONNOTE>
         <LABEL>
            <CONREF><![CDATA[ref01_008]]></CONREF>
         </LABEL>
         <MANIFEST>
            <CONREF><![CDATA[ref01_008]]></CONREF>
         </MANIFEST>
         <INVOICE>
            <CONREF><![CDATA[ref01_008]]></CONREF>
         </INVOICE>
         <EMAILTO><![CDATA[test.name@tnt.com]]></EMAILTO>
         <EMAILFROM><![CDATA[test.name@tnt.com]]></EMAILFROM>
      </PRINT>
```
It worked correct 

This method  ->createOptionalActivities() 

Added 

```
<PRINT><REQUIRED>
 <CONREF><![CDATA[ref01_008]]></CONREF>
</REQUIRED>
</PRINT>
```

Thanks

**Comment by 200MPH** [comment]
Hi,

Interesting.
What version you have please?

Following to TNT documentation that should work:
"To generate all possible shipment documents please specify the <REQUIRED> element."

**Comment by imjonos** [comment]
Following to TNT documentation that should work:
"To generate all possible shipment documents please specify the element."
----------------------
Yes but I tested it here   https://express.tnt.com/expresswebservices-website/app/shippingrequest.html?xmlfile=create_ship_book_print_shipment

And got the same response with empty Consignment Note

Now I using fork with my small fixies 

I used this in composer 
"200mph/tnt-express-connect": "^3.0.1",

**Comment by 200MPH** [comment]
Looks like I would have to append XML as you did.

Thanks!

**Comment by imjonos** [comment]
Great 
Thanks

**Comment by 200MPH** [comment]
Quick update.
The package has a small issue with handling HTTP errors.
In this case TNT server return error **500 Internal Error**, which was not handle by script.
You got  "Response XML document is not valid or empty" instead, which is not what we expected. Not a big issue, error was throw, but it content could misslead developers.
This will be fixed in next release.

Anyway...
When I called **getResults()** I've got a response from TNT instead error 500.
Their API throw 500 when you call any other function like getLabel().
So I'm going to improve that.
